rollup/internal/controller/sender/estimategas.go

```diff
@@ -10,24 +10,28 @@ import (
 	"github.com/scroll-tech/go-ethereum/log"
 )
 
-func (s *Sender) estimateLegacyGas(auth *bind.TransactOpts, contract *common.Address, value *big.Int, input []byte, minGasLimit uint64) (*FeeData, error) {
+func (s *Sender) estimateLegacyGas(auth *bind.TransactOpts, contract *common.Address, value *big.Int, input []byte, fallbackGasLimit uint64) (*FeeData, error) {
 	gasPrice, err := s.client.SuggestGasPrice(s.ctx)
 	if err != nil {
 		log.Error("estimateLegacyGas SuggestGasPrice failure", "error", err)
 		return nil, err
 	}
-	gasLimit, err := s.estimateGasLimit(auth, contract, input, gasPrice, nil, nil, value, minGasLimit)
+	gasLimit, err := s.estimateGasLimit(auth, contract, input, gasPrice, nil, nil, value)
 	if err != nil {
-		log.Error("estimateLegacyGas estimateGasLimit failure", "gasPrice", gasPrice, "error", err)
-		return nil, err
+		log.Error("estimateLegacyGas estimateGasLimit failure", "gas price", gasPrice, "from", auth.From.Hex(),
+			"nonce", auth.Nonce.Uint64(), "contract address", contract.Hex(), "fallback gas limit", fallbackGasLimit, "error", err)
+		if fallbackGasLimit == 0 {
+			return nil, err
+		}
+		gasLimit = fallbackGasLimit
 	}
 	return &FeeData{
 		gasPrice: gasPrice,
 		gasLimit: gasLimit,
 	}, nil
 }
 
-func (s *Sender) estimateDynamicGas(auth *bind.TransactOpts, contract *common.Address, value *big.Int, input []byte, minGasLimit uint64) (*FeeData, error) {
+func (s *Sender) estimateDynamicGas(auth *bind.TransactOpts, contract *common.Address, value *big.Int, input []byte, fallbackGasLimit uint64) (*FeeData, error) {
```

Comment:
Please also rename in `sender.SendTransaction`

Reply:
nice catch.

Reply:
fixed in [c849f5a](https://github.com/scroll-tech/scroll/pull/939/commits/c849f5adc782e2869ed1d0b9f1f549ba19dfb54a)